salt/modules/nilrt_ip.py

```diff
@@ -810,24 +742,9 @@ def set_dhcp_linklocal_all(interface):
         else:
             _restart(interface)
         return True
-    service = _interface_to_service(interface)
-    if not service:
-        if interface in map(lambda x: x.name, pyiface.getIfaces()):
-            return _enable_dhcp(interface)
-        raise salt.exceptions.CommandExecutionError('Invalid interface name: {0}'.format(interface))
-    service = pyconnman.ConnService(os.path.join(SERVICE_PATH, service))
-    ipv4 = service.get_property('IPv4.Configuration')
-    ipv4['Method'] = dbus.String('dhcp', variant_level=1)
-    ipv4['Address'] = dbus.String('', variant_level=1)
-    ipv4['Netmask'] = dbus.String('', variant_level=1)
-    ipv4['Gateway'] = dbus.String('', variant_level=1)
-    try:
-        service.set_property('IPv4.Configuration', ipv4)
-        service.set_property('Nameservers.Configuration', [''])  # reset nameservers list
-    except Exception as exc:
-        exc_msg = 'Couldn\'t set dhcp linklocal for service: {0}\nError: {1}\n'.format(service, exc)
-        raise salt.exceptions.CommandExecutionError(exc_msg)
-    return True
+    if interface in [x.name for x in pyiface.getIfaces()]:
+        return _change_dhcp_config(interface)
```

Comment:
is this setting the interface to dhcp-only? or dhcp-linklocal too?

Reply:
`dhcp-linklocal_all`
We don't have on NG:
* `dhcp_only`
* `linklocal_only`